salt/states/macpackage.py

```diff
@@ -232,5 +232,6 @@ def failed_pkg(f_pkg):
         if dmg:
             # Unmount to be kind
             __salt__["macpackage.unmount"](mount_point)
+            __salt__["cmd.run"]("rm -rf {}".format(mount_point))
```

Comment:
Is there a specific reason why you are shelling out?
I would prefer to use something native to python such as  `pathlib` 's `rmdir` call if the directory is expected to be empty or `shutil.rmtree`.